common/djangoapps/embargo/api.py

```diff
@@ -0,0 +1,266 @@
+        return REASONS['ip_country'].format(
+            ip_addr=ip_addr,
+            ip_country=ip_country,
+            from_course=_from_course_msg(course_id, course_is_embargoed)
+        )
+
+
+def _is_embargoed_by_profile_country(user, course_id="", course_is_embargoed=False):
+    """
+    Check whether the user is embargoed based on the country code in the user's profile.
+
+    Args:
+        user (User): The user attempting to access courseware.
+
+    Keyword Args:
+        course_id (unicode): The course the user is trying to access.
+        course_is_embargoed (boolean): Whether the course the user is accessing has been embargoed.
+
+    Returns:
+        A unicode message if the user is embargoed, otherwise `None`
+
+    """
+    cache_key = u'user.{user_id}.profile.country'.format(user_id=user.id)
+    profile_country = cache.get(cache_key)
+    if profile_country is None:
+        profile = getattr(user, 'profile', None)
+        if profile is not None and profile.country.code is not None:
+            profile_country = profile.country.code.upper()
+        else:
+            profile_country = ""
+        cache.set(cache_key, profile_country)
+
+    # with whitelist if returns false return error
+    if not CountryAccessRule.is_course_embargoed_in_country_list(course_id, profile_country, WHITE_LIST):
+        return REASONS['profile_country'].format(
+            user_id=unique_id_for_user(user),
+            profile_country=profile_country,
+            from_course=_from_course_msg(course_id, course_is_embargoed)
+        )
+
+    # with blacklist if returns true return error
+    if CountryAccessRule.is_course_embargoed_in_country_list(course_id, profile_country, BLACK_LIST):
+        return REASONS['profile_country'].format(
+            user_id=unique_id_for_user(user),
+            profile_country=profile_country,
+            from_course=_from_course_msg(course_id, course_is_embargoed)
+        )
+
+
+def _country_code_from_ip(ip_addr):
+    """
+    Return the country code associated with an IP address.
+    Handles both IPv4 and IPv6 addresses.
+
+    Args:
+        ip_addr (str): The IP address to look up.
+
+    Returns:
+        str: A 2-letter country code.
+
+    """
+    if ip_addr.find(':') >= 0:
+        return pygeoip.GeoIP(settings.GEOIPV6_PATH).country_code_by_addr(ip_addr)
+    else:
+        return pygeoip.GeoIP(settings.GEOIP_PATH).country_code_by_addr(ip_addr)
+
+
+def check_access(user, ip_address, course_key):
+    """
+    Check is the user with this ip_address has access to the given course
+
+    Params:
+        user (User): Currently logged in user object
+        ip_address (str): The ip_address of user
+        course_key (CourseLocator): CourseLocator object the user is trying to access
+
+    Returns:
+        Redirect to a URL configured in Django settings or a forbidden response if any constraints fails or None
```

Comment:
This function should return True or False, not a redirect.  In other contexts, we may want to check access without necessarily performing a redirect.  The middleware can check the return value and then perform the redirect.